### PR TITLE
bcachefs: Fix error in parsing of mount options

### PR DESCRIPTION
**Pull request by dlrobertson**
When parsing the mount options duplicate the given options. This is
required as the options are parsed twice and strsep is used in parsing.
The options will be modified into a possibly invalid options set for the
second round of parsing if the options are not duplicated before
parsing.

Signed-off-by: Dan Robertson <dan@dlrobertson.com>

Fixes: #223

### CODE REVIEW

File: fs/bcachefs/opts.c [hunk 1]
```diff
@@ -315,11 +315,21 @@ int bch2_opts_check_may_set(struct bch_fs *c)
...
 
-	while ((opt = strsep(&options, ",")) != NULL) {
+	if (!options)
+		return 0;
+
+	copied_opts = kmalloc(strlen(options) + 1, GFP_KERNEL);
+	if (!copied_opts)
+		return -1;
+
+	strcpy(copied_opts, options);
```
Comment (koverstreet) [inline fs/bcachefs/opts.c:330]: We can just use kstrdup here

File: fs/bcachefs/opts.c [hunk 2]
```diff
@@ -363,15 +373,19 @@ int bch2_parse_mount_opts(struct bch_fs *c, struct bch_opts *opts,
 		bch2_opt_set_by_id(opts, id, v);
 	}
 
+	kfree(copied_opts);
```
Comment (koverstreet) [inline fs/bcachefs/opts.c:376]: Instead of duplicating the kfree(), add an out label before this line, and then change the other returns to goto out

**Review by koverstreet — COMMENTED** [review]
Nice fix - couple small details, once those are done I'll merge it

**Comment by dlrobertson** [comment]
Thanks for the comments! I totally forgot about `kstrdup`

**Comment by koverstreet** [comment]
Thanks! Merged it into the testing branch.

**Pull request closed without merging**